### PR TITLE
ha: fix old cancel not call and adjust lint

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
we used too many `// make go vet happy`, and didn't call `oldCancel()`

### What is changed and how it works?
we can reuse old context 😂 

### Check List 

Tests 

 - pass original test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: pkg/ha/keepalive.go [hunk 1]
```diff
@@ -124,6 +124,7 @@ func KeepAlive(ctx context.Context, cli *clientv3.Client, workerName string, kee
 	}()
 
 	keepAliveCtx, keepAliveCancel := context.WithCancel(ctx)
+	_ = keepAliveCancel // make go vet happy
```
Comment (glorv) [inline pkg/ha/keepalive.go:127]:
```suggestion
	_ = keepAliveCancel // make go vet happy
	defer keepAliveCancel()
```

**Review by Ehco1996 — APPROVED** [review]
LGTM

File: pkg/ha/keepalive.go [hunk 2]
```diff
@@ -137,12 +135,10 @@ func KeepAlive(ctx context.Context, cli *clientv3.Client, workerName string, kee
 		case _, ok := <-ch:
 			if !ok {
 				log.L().Info("keep alive channel is closed")
-				keepAliveCancel() // make go vet happy
```
Comment (okJiang) [inline pkg/ha/keepalive.go:140]: what is `make go vet happy`?😂

**Review by okJiang — APPROVED** [review]

**Review by okJiang — APPROVED** [review]

**Pull request merged**